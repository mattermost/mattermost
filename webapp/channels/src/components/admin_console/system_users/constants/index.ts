// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum ColumnNames {
    username = 'usernameColumn',
    email = 'emailColumn',
    createAt = 'createAtColumn',
    lastLoginAt = 'lastLoginColumn',
    lastStatusAt = 'lastStatusAtColumn',
    lastPostDate = 'lastPostDateColumn',
    daysActive = 'daysActiveColumn',
    totalPosts = 'totalPostsColumn',
    actions = 'actionsColumn',
}

export enum StatusFilter {
    Any = 'any',
    Active = 'active',
    Deactivated = 'deactivated',
}

export enum RoleFilters {
    Any = 'any',
    Admin = 'system_admin',
    Member = 'system_user',
    Guest = 'system_guest',
}

export enum TeamFilters {
    AllTeams = 'teams_filter_for_all_teams',
    NoTeams = 'teams_filter_for_no_teams',
}
