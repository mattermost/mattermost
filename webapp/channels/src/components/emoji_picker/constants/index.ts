// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {defineMessage} from 'react-intl';

import type {EmojiCategory} from '@mattermost/types/emojis';

import * as Emoji from 'utils/emoji';

import type {Category, Categories} from '../types';

export const RECENT = 'recent';
export const SEARCH_RESULTS = 'searchResults';
export const SMILEY_EMOTION = 'smileys-emotion';
export const CUSTOM = 'custom';

const emojiCategories = {
    recent: {
        name: 'recent',
        label: defineMessage({
            id: 'emoji_picker.recent',
            defaultMessage: 'Recent',
        }),
        iconClassName: 'icon-clock-outline',
    },
    searchResults: {
        name: 'searchResults',
        label: defineMessage({
            id: 'emoji_picker.searchResults',
            defaultMessage: 'Search Results',
        }),
        iconClassName: '',
    },
    'smileys-emotion': {
        name: 'smileys-emotion',
        label: defineMessage({
            id: 'emoji_picker.smileys-emotion',
            defaultMessage: 'Smileys & Emotion',
        }),
        iconClassName: 'icon-emoticon-happy-outline',
    },
    'people-body': {
        name: 'people-body',
        label: defineMessage({
            id: 'emoji_picker.people-body',
            defaultMessage: 'People & Body',
        }),
        iconClassName: 'icon-account-outline',
    },
    'animals-nature': {
        name: 'animals-nature',
        label: defineMessage({
            id: 'emoji_picker.animals-nature',
            defaultMessage: 'Animals & Nature',
        }),
        iconClassName: 'icon-leaf-outline',
    },
    'food-drink': {
        name: 'food-drink',
        label: defineMessage({
            id: 'emoji_picker.food-drink',
            defaultMessage: 'Food & Drink',
        }),
        iconClassName: 'icon-food-apple',
    },
    'travel-places': {
        name: 'travel-places',
        label: defineMessage({
            id: 'emoji_picker.travel-places',
            defaultMessage: 'Travel & Places',
        }),
        iconClassName: 'icon-airplane-variant',
    },
    activities: {
        name: 'activities',
        label: defineMessage({
            id: 'emoji_picker.activities',
            defaultMessage: 'Activities',
        }),
        iconClassName: 'icon-basketball',
    },
    objects: {
        name: 'objects',
        label: defineMessage({
            id: 'emoji_picker.objects',
            defaultMessage: 'Objects',
        }),
        iconClassName: 'icon-lightbulb-outline',
    },
    symbols: {
        name: 'symbols',
        label: defineMessage({
            id: 'emoji_picker.symbols',
            defaultMessage: 'Symbols',
        }),
        iconClassName: 'icon-heart-outline',
    },
    flags: {
        name: 'flags',
        label: defineMessage({
            id: 'emoji_picker.flags',
            defaultMessage: 'Flags',
        }),
        iconClassName: 'icon-flag-outline',
    },
    custom: {
        name: 'custom',
        label: defineMessage({
            id: 'emoji_picker.custom',
            defaultMessage: 'Custom',
        }),
        iconClassName: 'icon-emoticon-custom-outline',
    },
} satisfies Record<EmojiCategory, Category>;

export const RECENT_EMOJI_CATEGORY: Pick<Categories, 'recent'> = {recent: emojiCategories.recent};
export const SEARCH_EMOJI_CATEGORY: Pick<Categories, typeof SEARCH_RESULTS> = {searchResults: emojiCategories.searchResults};

export const CATEGORIES: Categories = Emoji.CategoryNames.
    filter((category) => !(category === 'recent' || category === 'searchResults')).
    reduce((previousCategory, currentCategory) => {
        return {
            ...previousCategory,
            [currentCategory]: emojiCategories[currentCategory as EmojiCategory],
        };
    }, {} as Categories);

export const EMOJI_PER_ROW = 9; // needs to match variable `$emoji-per-row` in _variables.scss
export const ITEM_HEIGHT = 36; //as per .emoji-picker__item height in _emoticons.scss
export const EMOJI_CONTAINER_HEIGHT = 290; // If this changes, the spaceRequiredAbove and spaceRequiredBelow props passed to the EmojiPickerOverlay must be updated

export const CATEGORY_HEADER_ROW = 'categoryHeaderRow';
export const EMOJIS_ROW = 'emojisRow';

export const EMOJI_SCROLL_THROTTLE_DELAY = 150;
export const EMOJI_ROWS_OVERSCAN_COUNT = 1;

export const CUSTOM_EMOJIS_PER_PAGE = 200;
export const CUSTOM_EMOJI_SEARCH_THROTTLE_TIME_MS = 1000;
