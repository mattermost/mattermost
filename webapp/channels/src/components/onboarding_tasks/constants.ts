// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const FINISHED = 999;

export const OnboardingTaskCategory = 'onboarding_task_list';

// Whole task list is based on these
export const OnboardingTasksName = {
    CHANNELS_TOUR: 'channels_tour',
    INVITE_PEOPLE: 'invite_people',
    DOWNLOAD_APP: 'download_app',
    COMPLETE_YOUR_PROFILE: 'complete_your_profile',
    VISIT_SYSTEM_CONSOLE: 'visit_system_console',
    START_TRIAL: 'start_trial',
};

export const OnboardingTaskList = {
    ONBOARDING_TASK_LIST_OPEN: 'onboarding_task_list_open',
    ONBOARDING_TASK_LIST_SHOW: 'onboarding_task_list_show',
    ONBOARDING_TASK_LIST_CLOSE: 'onboarding_task_list_close',
    DECLINED_ONBOARDING_TASK_LIST: 'declined_onboarding_task_list',
};

export const GenericTaskSteps = {
    START: 0,
    STARTED: 1,
    FINISHED,
};

export const TaskNameMapToSteps = {
    [OnboardingTasksName.CHANNELS_TOUR]: GenericTaskSteps,
    [OnboardingTasksName.COMPLETE_YOUR_PROFILE]: GenericTaskSteps,
    [OnboardingTasksName.DOWNLOAD_APP]: GenericTaskSteps,
    [OnboardingTasksName.VISIT_SYSTEM_CONSOLE]: GenericTaskSteps,
    [OnboardingTasksName.INVITE_PEOPLE]: GenericTaskSteps,
    [OnboardingTasksName.START_TRIAL]: GenericTaskSteps,
};

