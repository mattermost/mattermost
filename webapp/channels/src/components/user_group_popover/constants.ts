// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export enum Load {
    DONE,
    LOADING,
    FAILED,
}

export const USER_GROUP_POPOVER_OPENING_DELAY = 300;
export const USER_GROUP_POPOVER_CLOSING_DELAY = 500;
