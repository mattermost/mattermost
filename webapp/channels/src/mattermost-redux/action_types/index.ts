// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import keyMirror from 'mattermost-redux/utils/key_mirror';

export const ActionTypes = keyMirror({
    GET_SHARED_CHANNELS: null,
    RECEIVED_SHARED_CHANNELS_WITH_REMOTES: null,
});