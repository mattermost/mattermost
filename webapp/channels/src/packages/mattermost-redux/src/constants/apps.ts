// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {AppCallResponseType, AppExpandLevel, AppFieldType} from '@mattermost/types/apps';

export const AppBindingLocations = {
    POST_MENU_ITEM: '/post_menu',
    CHANNEL_HEADER_ICON: '/channel_header',
    APP_BAR: '/app_bar',
    COMMAND: '/command',
    IN_POST: '/in_post',
    MARKETPLACE: '/marketplace',
};

export const AppBindingPresentations = {
    MODAL: 'modal',
};

export const AppCallResponseTypes: { [name: string]: AppCallResponseType } = {
    OK: 'ok',
    ERROR: 'error',
    FORM: 'form',
    CALL: 'call',
    NAVIGATE: 'navigate',
};

export const AppExpandLevels: { [name: string]: AppExpandLevel } = {
    EXPAND_DEFAULT: '',
    EXPAND_NONE: 'none',
    EXPAND_ALL: 'all',
    EXPAND_SUMMARY: 'summary',
};

export const AppFieldTypes: { [name: string]: AppFieldType } = {
    TEXT: 'text',
    STATIC_SELECT: 'static_select',
    DYNAMIC_SELECT: 'dynamic_select',
    BOOL: 'bool',
    USER: 'user',
    CHANNEL: 'channel',
    MARKDOWN: 'markdown',
    RADIO: 'radio',
};
