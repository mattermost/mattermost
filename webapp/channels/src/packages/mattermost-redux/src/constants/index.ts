// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import Emoji from './emoji';
import Files from './files';
import General from './general';
import Permissions from './permissions';
import Plugins from './plugins';
import Posts from './posts';
import Preferences from './preferences';
import RequestStatus from './request_status';
import Roles from './roles';
import Stats from './stats';
import Teams from './teams';
import Threads from './threads';
import Users from './users';
import WebsocketEvents from './websocket';

export {General, Preferences, Posts, Files, RequestStatus, WebsocketEvents, Teams, Stats, Permissions, Emoji, Plugins, Users, Roles, Threads};
