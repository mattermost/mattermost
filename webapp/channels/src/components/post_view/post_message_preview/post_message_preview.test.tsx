// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import type {ChannelType} from '@mattermost/types/channels';
import type {Post, PostEmbed} from '@mattermost/types/posts';
import type {UserProfile} from '@mattermost/types/users';

import {General} from 'mattermost-redux/constants';

import PostMessagePreview from './post_message_preview';
import type {Props} from './post_message_preview';

jest.mock('components/properties_card_view/propertyValueRenderer/post_preview_property_renderer/post_preview_property_renderer', () => {
    return jest.fn(() => <div data-testid='post-preview-property-renderer-mock'>{'PostPreviewPropertyRenderer Mock'}</div>);
});

describe('PostMessagePreview', () => {
    const previewPost = {
        id: 'post_id',
        message: 'post message',
        metadata: {},
    } as Post;

    const user = {
        id: 'user_1',
        username: 'username1',
    } as UserProfile;

    const baseProps: Props = {
        metadata: {
            channel_display_name: 'channel name',
            team_name: 'team1',
            post_id: 'post_id',
            channel_type: 'O',
            channel_id: 'channel_id',
        },
        previewPost,
        user,
        hasImageProxy: false,
        enablePostIconOverride: false,
        isEmbedVisible: true,
        compactDisplay: false,
        currentTeamUrl: 'team1',
        channelDisplayName: 'channel name',
        handleFileDropdownOpened: jest.fn(),
        actions: {
            toggleEmbedVisibility: jest.fn(),
        },
        isPostPriorityEnabled: false,
    };

    test('should render correctly', () => {
        const wrapper = shallow(<PostMessagePreview {...baseProps}/>);

        expect(wrapper).toMatchSnapshot();
    });

    test('should render without preview', () => {
        const wrapper = shallow(
            <PostMessagePreview
                {...baseProps}
                previewPost={undefined}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('show render without preview when preview posts becomes undefined after being defined', () => {
        const props = {...baseProps};
        let wrapper = shallow(
            <PostMessagePreview
                {...props}
            />,
        );

        expect(wrapper).toMatchSnapshot();
        let permalink = wrapper.find('.permalink');
        expect(permalink.length).toBe(1);

        // now we'll set the preview post to undefined. This happens when the
        // previewed post is deleted.
        props.previewPost = undefined;

        wrapper = shallow(
            <PostMessagePreview
                {...props}
            />,
        );
        expect(wrapper).toMatchSnapshot();
        permalink = wrapper.find('.permalink');
        expect(permalink.length).toBe(0);
    });

    test('should not render bot icon', () => {
        const postProps = {
            override_icon_url: 'https://fakeicon.com/image.jpg',
            use_user_icon: 'false',
            from_webhook: 'false',
        };

        const postPreview = {
            ...previewPost,
            props: postProps,
        } as unknown as Post;

        const props = {
            ...baseProps,
            previewPost: postPreview,
        };
        const wrapper = shallow(
            <PostMessagePreview
                {...props}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should render bot icon', () => {
        const postProps = {
            override_icon_url: 'https://fakeicon.com/image.jpg',
            use_user_icon: false,
            from_webhook: 'true',
        };

        const postPreview = {
            ...previewPost,
            props: postProps,
        } as unknown as Post;

        const props = {
            ...baseProps,
            previewPost: postPreview,
            enablePostIconOverride: true,
        };
        const wrapper = shallow(
            <PostMessagePreview
                {...props}
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    describe('nested previews', () => {
        const files = {
            file_ids: [
                'file_1',
                'file_2',
            ],
        };

        const opengraphMetadata = {
            type: 'opengraph',
            url: 'https://example.com',
        } as PostEmbed;

        test('should render opengraph preview', () => {
            const postPreview = {
                ...previewPost,
                metadata: {
                    embeds: [opengraphMetadata],
                },
            } as Post;

            const props = {
                ...baseProps,
                previewPost: postPreview,
            };

            const wrapper = shallow(<PostMessagePreview {...props}/>);

            expect(wrapper).toMatchSnapshot();
        });

        test('should render file preview', () => {
            const postPreview = {
                ...previewPost,
                ...files,
            } as Post;

            const props = {
                ...baseProps,
                previewPost: postPreview,
            };

            const wrapper = shallow(<PostMessagePreview {...props}/>);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('direct and group messages', () => {
        const channelTypes = [General.DM_CHANNEL, General.GM_CHANNEL] as ChannelType[];

        test.each(channelTypes)('should render preview for %s message', (type) => {
            const metadata = {
                ...baseProps.metadata,
                team_name: '',
                channel_type: type,
                channel_id: 'channel_id',
            };

            const props = {
                ...baseProps,
                metadata,
            };

            const wrapper = shallow(
                <PostMessagePreview
                    {...props}
                />,
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
