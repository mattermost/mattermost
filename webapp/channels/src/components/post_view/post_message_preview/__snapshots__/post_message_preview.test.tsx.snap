// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostMessagePreview direct and group messages should render preview for D message 1`] = `
<PostAttachmentContainer
  className="permalink"
  link="team1/pl/post_id"
>
  <div
    className="post-preview"
  >
    <div
      className="post-preview__header"
    >
      <div
        className="col col__name"
      >
        <div
          className="post__img"
        >
          <span
            className="profile-icon"
          >
            <PreviewPostAvatar
              enablePostIconOverride={false}
              hasImageProxy={false}
              post={
                Object {
                  "id": "post_id",
                  "message": "post message",
                  "metadata": Object {},
                }
              }
              user={
                Object {
                  "id": "user_1",
                  "username": "username1",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="col col__name permalink--username"
      >
        <Connect(UserProfile)
          disablePopover={true}
          overwriteName=""
          userId="user_1"
        />
      </div>
      <div
        className="col d-flex align-items-center"
      >
        <Connect(injectIntl(Timestamp))
          className="post-preview__time"
          day="numeric"
          units={
            Array [
              "now",
              "minute",
              "hour",
              "day",
            ]
          }
          useTime={false}
        />
      </div>
    </div>
    <Connect(PostMessageView)
      maxHeight={105}
      overflowType="ellipsis"
      post={
        Object {
          "id": "post_id",
          "message": "post message",
          "metadata": Object {},
        }
      }
    />
    <div
      className="post__preview-footer"
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Only visible to users in ~{channel}"
          id="post_message_preview.channel"
          values={
            Object {
              "channel": "channel name",
            }
          }
        />
      </p>
    </div>
  </div>
</PostAttachmentContainer>
`;

exports[`PostMessagePreview direct and group messages should render preview for G message 1`] = `
<PostAttachmentContainer
  className="permalink"
  link="team1/pl/post_id"
>
  <div
    className="post-preview"
  >
    <div
      className="post-preview__header"
    >
      <div
        className="col col__name"
      >
        <div
          className="post__img"
        >
          <span
            className="profile-icon"
          >
            <PreviewPostAvatar
              enablePostIconOverride={false}
              hasImageProxy={false}
              post={
                Object {
                  "id": "post_id",
                  "message": "post message",
                  "metadata": Object {},
                }
              }
              user={
                Object {
                  "id": "user_1",
                  "username": "username1",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="col col__name permalink--username"
      >
        <Connect(UserProfile)
          disablePopover={true}
          overwriteName=""
          userId="user_1"
        />
      </div>
      <div
        className="col d-flex align-items-center"
      >
        <Connect(injectIntl(Timestamp))
          className="post-preview__time"
          day="numeric"
          units={
            Array [
              "now",
              "minute",
              "hour",
              "day",
            ]
          }
          useTime={false}
        />
      </div>
    </div>
    <Connect(PostMessageView)
      maxHeight={105}
      overflowType="ellipsis"
      post={
        Object {
          "id": "post_id",
          "message": "post message",
          "metadata": Object {},
        }
      }
    />
    <div
      className="post__preview-footer"
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Only visible to users in ~{channel}"
          id="post_message_preview.channel"
          values={
            Object {
              "channel": "channel name",
            }
          }
        />
      </p>
    </div>
  </div>
</PostAttachmentContainer>
`;

exports[`PostMessagePreview nested previews should render file preview 1`] = `
<PostAttachmentContainer
  className="permalink"
  link="/team1/pl/post_id"
>
  <div
    className="post-preview"
  >
    <div
      className="post-preview__header"
    >
      <div
        className="col col__name"
      >
        <div
          className="post__img"
        >
          <span
            className="profile-icon"
          >
            <PreviewPostAvatar
              enablePostIconOverride={false}
              hasImageProxy={false}
              post={
                Object {
                  "file_ids": Array [
                    "file_1",
                    "file_2",
                  ],
                  "id": "post_id",
                  "message": "post message",
                  "metadata": Object {},
                }
              }
              user={
                Object {
                  "id": "user_1",
                  "username": "username1",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="col col__name permalink--username"
      >
        <Connect(UserProfile)
          disablePopover={true}
          overwriteName=""
          userId="user_1"
        />
      </div>
      <div
        className="col d-flex align-items-center"
      >
        <Connect(injectIntl(Timestamp))
          className="post-preview__time"
          day="numeric"
          units={
            Array [
              "now",
              "minute",
              "hour",
              "day",
            ]
          }
          useTime={false}
        />
      </div>
    </div>
    <Connect(PostMessageView)
      maxHeight={105}
      overflowType="ellipsis"
      post={
        Object {
          "file_ids": Array [
            "file_1",
            "file_2",
          ],
          "id": "post_id",
          "message": "post message",
          "metadata": Object {},
        }
      }
    />
    <Connect(FileAttachmentList)
      compactDisplay={false}
      handleFileDropdownOpened={[MockFunction]}
      isInPermalink={true}
      post={
        Object {
          "file_ids": Array [
            "file_1",
            "file_2",
          ],
          "id": "post_id",
          "message": "post message",
          "metadata": Object {},
        }
      }
    />
    <div
      className="post__preview-footer"
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Only visible to users in ~{channel}"
          id="post_message_preview.channel"
          values={
            Object {
              "channel": "channel name",
            }
          }
        />
      </p>
    </div>
  </div>
</PostAttachmentContainer>
`;

exports[`PostMessagePreview nested previews should render opengraph preview 1`] = `
<PostAttachmentContainer
  className="permalink"
  link="/team1/pl/post_id"
>
  <div
    className="post-preview"
  >
    <div
      className="post-preview__header"
    >
      <div
        className="col col__name"
      >
        <div
          className="post__img"
        >
          <span
            className="profile-icon"
          >
            <PreviewPostAvatar
              enablePostIconOverride={false}
              hasImageProxy={false}
              post={
                Object {
                  "id": "post_id",
                  "message": "post message",
                  "metadata": Object {
                    "embeds": Array [
                      Object {
                        "type": "opengraph",
                        "url": "https://example.com",
                      },
                    ],
                  },
                }
              }
              user={
                Object {
                  "id": "user_1",
                  "username": "username1",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="col col__name permalink--username"
      >
        <Connect(UserProfile)
          disablePopover={true}
          overwriteName=""
          userId="user_1"
        />
      </div>
      <div
        className="col d-flex align-items-center"
      >
        <Connect(injectIntl(Timestamp))
          className="post-preview__time"
          day="numeric"
          units={
            Array [
              "now",
              "minute",
              "hour",
              "day",
            ]
          }
          useTime={false}
        />
      </div>
    </div>
    <Connect(PostMessageView)
      maxHeight={105}
      overflowType="ellipsis"
      post={
        Object {
          "id": "post_id",
          "message": "post message",
          "metadata": Object {
            "embeds": Array [
              Object {
                "type": "opengraph",
                "url": "https://example.com",
              },
            ],
          },
        }
      }
    />
    <Connect(PostAttachmentOpenGraph)
      isEmbedVisible={true}
      isInPermalink={true}
      link="https://example.com"
      post={
        Object {
          "id": "post_id",
          "message": "post message",
          "metadata": Object {
            "embeds": Array [
              Object {
                "type": "opengraph",
                "url": "https://example.com",
              },
            ],
          },
        }
      }
      postId="post_id"
      toggleEmbedVisibility={[Function]}
    />
    <div
      className="post__preview-footer"
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Only visible to users in ~{channel}"
          id="post_message_preview.channel"
          values={
            Object {
              "channel": "channel name",
            }
          }
        />
      </p>
    </div>
  </div>
</PostAttachmentContainer>
`;

exports[`PostMessagePreview should not render bot icon 1`] = `
<PostAttachmentContainer
  className="permalink"
  link="/team1/pl/post_id"
>
  <div
    className="post-preview"
  >
    <div
      className="post-preview__header"
    >
      <div
        className="col col__name"
      >
        <div
          className="post__img"
        >
          <span
            className="profile-icon"
          >
            <PreviewPostAvatar
              enablePostIconOverride={false}
              hasImageProxy={false}
              post={
                Object {
                  "id": "post_id",
                  "message": "post message",
                  "metadata": Object {},
                  "props": Object {
                    "from_webhook": "false",
                    "override_icon_url": "https://fakeicon.com/image.jpg",
                    "use_user_icon": "false",
                  },
                }
              }
              user={
                Object {
                  "id": "user_1",
                  "username": "username1",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="col col__name permalink--username"
      >
        <Connect(UserProfile)
          disablePopover={true}
          overwriteName=""
          userId="user_1"
        />
      </div>
      <div
        className="col d-flex align-items-center"
      >
        <Connect(injectIntl(Timestamp))
          className="post-preview__time"
          day="numeric"
          units={
            Array [
              "now",
              "minute",
              "hour",
              "day",
            ]
          }
          useTime={false}
        />
      </div>
    </div>
    <Connect(PostMessageView)
      maxHeight={105}
      overflowType="ellipsis"
      post={
        Object {
          "id": "post_id",
          "message": "post message",
          "metadata": Object {},
          "props": Object {
            "from_webhook": "false",
            "override_icon_url": "https://fakeicon.com/image.jpg",
            "use_user_icon": "false",
          },
        }
      }
    />
    <div
      className="post__preview-footer"
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Only visible to users in ~{channel}"
          id="post_message_preview.channel"
          values={
            Object {
              "channel": "channel name",
            }
          }
        />
      </p>
    </div>
  </div>
</PostAttachmentContainer>
`;

exports[`PostMessagePreview should render bot icon 1`] = `
<PostAttachmentContainer
  className="permalink"
  link="/team1/pl/post_id"
>
  <div
    className="post-preview"
  >
    <div
      className="post-preview__header"
    >
      <div
        className="col col__name"
      >
        <div
          className="post__img"
        >
          <span
            className="profile-icon"
          >
            <PreviewPostAvatar
              enablePostIconOverride={true}
              hasImageProxy={false}
              post={
                Object {
                  "id": "post_id",
                  "message": "post message",
                  "metadata": Object {},
                  "props": Object {
                    "from_webhook": "true",
                    "override_icon_url": "https://fakeicon.com/image.jpg",
                    "use_user_icon": false,
                  },
                }
              }
              user={
                Object {
                  "id": "user_1",
                  "username": "username1",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="col col__name permalink--username"
      >
        <Connect(UserProfile)
          disablePopover={true}
          overwriteName=""
          userId="user_1"
        />
      </div>
      <div
        className="col d-flex align-items-center"
      >
        <Connect(injectIntl(Timestamp))
          className="post-preview__time"
          day="numeric"
          units={
            Array [
              "now",
              "minute",
              "hour",
              "day",
            ]
          }
          useTime={false}
        />
      </div>
    </div>
    <Connect(PostMessageView)
      maxHeight={105}
      overflowType="ellipsis"
      post={
        Object {
          "id": "post_id",
          "message": "post message",
          "metadata": Object {},
          "props": Object {
            "from_webhook": "true",
            "override_icon_url": "https://fakeicon.com/image.jpg",
            "use_user_icon": false,
          },
        }
      }
    />
    <div
      className="post__preview-footer"
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Only visible to users in ~{channel}"
          id="post_message_preview.channel"
          values={
            Object {
              "channel": "channel name",
            }
          }
        />
      </p>
    </div>
  </div>
</PostAttachmentContainer>
`;

exports[`PostMessagePreview should render correctly 1`] = `
<PostAttachmentContainer
  className="permalink"
  link="/team1/pl/post_id"
>
  <div
    className="post-preview"
  >
    <div
      className="post-preview__header"
    >
      <div
        className="col col__name"
      >
        <div
          className="post__img"
        >
          <span
            className="profile-icon"
          >
            <PreviewPostAvatar
              enablePostIconOverride={false}
              hasImageProxy={false}
              post={
                Object {
                  "id": "post_id",
                  "message": "post message",
                  "metadata": Object {},
                }
              }
              user={
                Object {
                  "id": "user_1",
                  "username": "username1",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="col col__name permalink--username"
      >
        <Connect(UserProfile)
          disablePopover={true}
          overwriteName=""
          userId="user_1"
        />
      </div>
      <div
        className="col d-flex align-items-center"
      >
        <Connect(injectIntl(Timestamp))
          className="post-preview__time"
          day="numeric"
          units={
            Array [
              "now",
              "minute",
              "hour",
              "day",
            ]
          }
          useTime={false}
        />
      </div>
    </div>
    <Connect(PostMessageView)
      maxHeight={105}
      overflowType="ellipsis"
      post={
        Object {
          "id": "post_id",
          "message": "post message",
          "metadata": Object {},
        }
      }
    />
    <div
      className="post__preview-footer"
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Only visible to users in ~{channel}"
          id="post_message_preview.channel"
          values={
            Object {
              "channel": "channel name",
            }
          }
        />
      </p>
    </div>
  </div>
</PostAttachmentContainer>
`;

exports[`PostMessagePreview should render without preview 1`] = `""`;

exports[`PostMessagePreview show render without preview when preview posts becomes undefined after being defined 1`] = `
<PostAttachmentContainer
  className="permalink"
  link="/team1/pl/post_id"
>
  <div
    className="post-preview"
  >
    <div
      className="post-preview__header"
    >
      <div
        className="col col__name"
      >
        <div
          className="post__img"
        >
          <span
            className="profile-icon"
          >
            <PreviewPostAvatar
              enablePostIconOverride={false}
              hasImageProxy={false}
              post={
                Object {
                  "id": "post_id",
                  "message": "post message",
                  "metadata": Object {},
                }
              }
              user={
                Object {
                  "id": "user_1",
                  "username": "username1",
                }
              }
            />
          </span>
        </div>
      </div>
      <div
        className="col col__name permalink--username"
      >
        <Connect(UserProfile)
          disablePopover={true}
          overwriteName=""
          userId="user_1"
        />
      </div>
      <div
        className="col d-flex align-items-center"
      >
        <Connect(injectIntl(Timestamp))
          className="post-preview__time"
          day="numeric"
          units={
            Array [
              "now",
              "minute",
              "hour",
              "day",
            ]
          }
          useTime={false}
        />
      </div>
    </div>
    <Connect(PostMessageView)
      maxHeight={105}
      overflowType="ellipsis"
      post={
        Object {
          "id": "post_id",
          "message": "post message",
          "metadata": Object {},
        }
      }
    />
    <div
      className="post__preview-footer"
    >
      <p>
        <MemoizedFormattedMessage
          defaultMessage="Only visible to users in ~{channel}"
          id="post_message_preview.channel"
          values={
            Object {
              "channel": "channel name",
            }
          }
        />
      </p>
    </div>
  </div>
</PostAttachmentContainer>
`;

exports[`PostMessagePreview show render without preview when preview posts becomes undefined after being defined 2`] = `""`;
